File: Python/project/app/modules/database_module/models/default/offer.py
from tortoise import fields

from app.modules.database_module.models.database_model import DatabaseModel
from app.modules.database_module.models.offer_type import OfferType


class Offer(DatabaseModel):
    title = fields.CharField(max_length=255)
    description = fields.TextField()
    vacancies_number = fields.IntField()
    start_date = fields.DateField()
    end_date = fields.DateField(null=True)
    type: OfferType = fields.IntField()

    address = fields.CharField(max_length=100)
    postal_code = fields.CharField(max_length=10)

    # Contact
    contact_name = fields.CharField(max_length=100)
    contact_email = fields.CharField(max_length=255)
    contact_phone = fields.CharField(max_length=20, null=True)

    # Relations
    company = fields.ForeignKeyField("default.Company", related_name="offers")
    degree = fields.ForeignKeyField("default.Degree", related_name="degree")

File: Kotlin/domain/src/main/java/com/mobivery/template/domain/model/offer/Offer.kt
import java.time.LocalDate

data class Offer(
    val id: Int,
    val title: String,
    val description: String? = null,
    val vacancies: Int,
    val startDate: LocalDate,
    val endDate: LocalDate? = null,
    val views: Int,
    val type: Int,
    val address: String,
    val postalCode: String,
    val contactName: String,
    val contactEmail: String,
    val contactPhone: String? = null,
    val company: String,
    val degreeId: String,
    val degreeName: String,
    val companyPhoto: String,
    val inscribe: Boolean?,
    val presentationCard: String? = null,
    val inscriptionsCandidacy: Int,
)
